cdap-client/src/main/java/co/cask/cdap/client/MetadataClient.java

```diff
@@ -60,7 +60,7 @@ protected HttpResponse execute(HttpRequest request, int... allowedErrorCodes)
 
   @Override
   protected URL resolve(Id.Namespace namespace, String resource) throws MalformedURLException {
```

Comment:
remove this method?

Reply:
this has not been deprecated, so i am not sure about removing this now. ideally we need to have resolve with `NamespaceId` instead of `Id.Namespace` and use that.

Reply:
Does the corresponding new method already exist? Or was that also missed back then?

Reply:
it doesn't exist, that was also missed.

Reply:
do we have the same deprecation policies for cdap-client? Do we have to deprecate before removing?

Reply:
after discussion we decided to replace with `NamespaceId`. As there is not a similar usage of  `Id` without being marked as deprecated, in cdap-client and cdap-api. 

in cdap-proto the usages are around translation from old entity id to new entity id and codecs, so they can be removed as well. i will do that in a separate PR.